codespell_lib/tests/test_basic.py

```diff
@@ -97,6 +97,9 @@ def test_basic(
     with fname.open("a") as f:
         f.write("this is a test file\n")
     assert cs.main(fname) == 0, "good"
+    with fname.open("a") as f:
+        f.write("var = '\\nDoes not error on newline'\n")
```

Comment:
This test relied on ndoes being in the dictionary, which I don't think it is.

I also found when adding a positive test that it doesn't match/trigger anyway because of the apostrophe at the start.

I've added some more thorough tests in #2879 .